config/initializers/http_client_proxy.rb

```diff
@@ -0,0 +1,26 @@
+Rails.application.configure do
+  config.x.http_client_proxy = {}
+  ['HTTP_PROXY', 'http_proxy'].each do |name|
+    next if ENV[name].blank?
+    proxy = URI.parse(ENV[name])
+    raise Mastodon::ValidationError, "Unsupported proxy type: #{proxy.scheme}" unless ["http", "https"].include? proxy.scheme
```

Comment:
This error is meant for application start, so no need to use Mastodon::ValidationError. You can just raise an anonymous exception.